exercises/practice/forth/.meta/example.awk

```diff
@@ -0,0 +1,175 @@
+#!/usr/bin/env gawk -f
+
+BEGINFILE {
+    delete stack
+    s = 0
+    delete macros
+}
+
+ENDFILE {
+    print join(stack, 1, s)
+}
+
+############################################################
+# process the file
+
+{
+    $0 = toupper($0)
+    $1 = $1
+}
+
+$1 == ":" {
+    record_macro()
+    next
+}
+
+{
+    for (i = 1; i <= NF; i++) {
+        if ($i in macros) {
+            $i = macros[$i]
+            rewrite_0()
+        }
+
+        if      (is_integer($i))  push($i)
+        else if ($i ~ /^[-+*/]$/) arithmetic($i)
+        else if ($i == "DROP")    drop()
+        else if ($i == "SWAP")    swap()
+        else if ($i == "DUP")     dup()
+        else if ($i == "OVER")    over()
+        else die("undefined operation")
+    }
+}
+
+############################################################
+# stack functions
+
+function push(value) {
+    stack[++s] = value
+}
+
+function pop(    value) {
```

Comment:
```suggestion
function pop(value) {
```

Reply:
The extra whitespace is awk convention. The only concept of "local variables" awk has are function parameters.

From [Function Definition Syntax](https://www.gnu.org/software/gawk/manual/html_node/Definition-Syntax.html)
> Usually when you write a function, you know how many names you intend to use for arguments and how many you intend to use as local variables. It is conventional to place some extra space between the arguments and the local variables, in order to document how your function is supposed to be used.